File: packages/ruby-sdk/lib/strait/operations/stats.rb
# frozen_string_literal: true

module Strait
  module Operations
    # Service for platform statistics.
    class StatsService < BaseService
      # List platform statistics.
      # GET /v1/stats
      def list(query: nil)
        _request(:get, "/v1/stats", query: query)
      end
    end
  end
end

File: packages/python-sdk/strait/operations/stats.py
"""Queue statistics operations."""

from __future__ import annotations

from typing import Any

from strait.operations._base import AsyncBaseService, BaseService


class StatsService(BaseService):
    def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return self._request("GET", "/v1/stats", query=query)


class AsyncStatsService(AsyncBaseService):
    async def list(self, *, query: dict[str, str] | None = None) -> dict[str, Any]:
        return await self._request("GET", "/v1/stats", query=query)
